src/components/CCPDashboard/PatientDetailsDialog.tsx

```diff
@@ -34,6 +50,73 @@ interface PatientDetail {
   styles?: object;
 }
 
+export const PatientDetailsDialogRouting = ({
+  match: {
+    params: { eventId, patientId, ccpId },
+  },
+}: {
+  match: {
+    params: {
+      eventId: string;
+      patientId?: string;
+      ccpId: string;
+    };
+  };
+}) => {
+  const [openDetails, setOpenDetails] = React.useState(true);
+  const history = useHistory();
+  const classes = useStyles();
+
+  const { data } = useQuery(
+    patientId ? GET_PATIENT_BY_ID(patientId) : GET_ALL_PATIENTS
+  );
+
+  const patient: Patient = data ? data.patient : null;
+
+  const handleCloseDetails = () => {
+    setOpenDetails(false);
+    history.push(`/events/${eventId}/ccps/${ccpId}`);
+  };
+
+  return (
+    <div>
+      {patient ? (
```

Comment:
can you explain why we need to have this check for `patient` and why we want to redirect to `/events` if it's null/falsy? i'm just unsure when `data.patient` will come out as null - can you describe a scenario when that might happen?